src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetApprenticeshipsValidate/GetApprenticeshipsValidateQueryHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+
+namespace SFA.DAS.CommitmentsV2.Application.Queries.GetApprenticeshipsValidate
+{
+    public class GetApprenticeshipsValidateQueryHandler : IRequestHandler<GetApprenticeshipsValidateQuery, GetApprenticeshipsValidateQueryResult>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+
+        public GetApprenticeshipsValidateQueryHandler(Lazy<ProviderCommitmentsDbContext> dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<GetApprenticeshipsValidateQueryResult> Handle(GetApprenticeshipsValidateQuery request, CancellationToken cancellationToken)
+        {
+            var result = await _dbContext.Value
+                .Apprenticeships
+                .Include(a => a.Cohort)
+                .ThenInclude(c => c.AccountLegalEntity)
+                .Where(a => a.FirstName == request.FirstName && a.LastName == request.LastName && a.DateOfBirth == request.DateOfBirth)
```

Comment:
It would be worth looking at whether an index is required for this.

Reply:
Done